pkg/auth/oidc/client_credentials.go

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package oidc
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/Peripli/service-manager-cli/internal/util"
+	"github.com/Peripli/service-manager-cli/pkg/auth"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+// ClientCredentialsStrategy implements client credentials flow authentication strategy
+type ClientCredentialsStrategy struct {
+	*clientcredentials.Config
+	httpClient *http.Client
+}
+
+// NewClientCredentialsStrategy returns client credentials auth strategy
+func NewClientCredentialsStrategy(options *auth.Options) (auth.AuthenticationStrategy, *auth.Options, error) {
```

Comment:
This looks very similar to oidc. NewOpenIDStrategy. Also the authenticate methods are pretty similar with the exception that one case you do `s.token` and it the other s. PasswordCredentialsToken`.  So two files for 1 line of code difference seems like an avoidable code duplication. Maybe we might even consider dropping the strategy indirection (abstraction) in general